controller/MiNiFiController.cpp

```diff
@@ -15,57 +15,52 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#include <fcntl.h>
-#include <cstdio>
-#include <semaphore.h>
-#include <csignal>
 #include <vector>
-#include <queue>
-#include <map>
 #include <iostream>
 
-#include "core/Core.h"
-
-#include "core/FlowConfiguration.h"
-#include "core/ConfigurationFactory.h"
-#include "core/RepositoryFactory.h"
-#include "FlowController.h"
 #include "MainHelper.h"
 #include "properties/Configure.h"
 #include "Controller.h"
 #include "c2/ControllerSocketProtocol.h"
+#include "core/controller/ControllerService.h"
+#include "core/extension/ExtensionManager.h"
+#include "io/StreamFactory.h"
+#include "core/ConfigurationFactory.h"
 
 #include "cxxopts.hpp"
 
 namespace minifi = org::apache::nifi::minifi;
 
-int main(int argc, char **argv) {
-  const auto logger = minifi::core::logging::LoggerConfiguration::getConfiguration().getLogger("controller");
+std::shared_ptr<minifi::core::controller::ControllerService> getControllerService(const std::shared_ptr<minifi::Configure> &configuration,
+    const std::string &service_name, const std::string& minifi_home) {
+  std::string nifi_configuration_class_name = "yamlconfiguration";
```

Comment:
Good point, did not change it after rebasing, updated in a582bd90b8f188e1cc20bc2e9e81d358250cb230

Reply:
line 44 needs to be updated, too, to use `nifi_flow_configuration_file` instead of `"$HOME/conf/config.yml"`

Reply:
Are you sure `nullptr` is an OK value for `ConfigurationContext::filesystem`?  The default value is `std::make_shared<utils::file::FileSystem>()`, so I'm wondering if it is assumed to be non-null somewhere.

Reply:
Although I think it should not be used in this case, it may be better to stick with the default value, updated in 8735105d4cd70f6bbe33fbe43bd17760382a9b1d